homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,265 @@
+"""samsungctl and samsungtvws bridge classes."""
+from abc import ABC, abstractmethod
+
+from samsungctl import Remote
+from samsungctl.exceptions import AccessDenied, ConnectionClosed, UnhandledResponse
+from samsungtvws import SamsungTVWS
+from samsungtvws.exceptions import ConnectionFailure
+from websocket import WebSocketException
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_ID,
+    CONF_METHOD,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TIMEOUT,
+    CONF_TOKEN,
+)
+
+from .const import (
+    CONF_DESCRIPTION,
+    LOGGER,
+    METHOD_LEGACY,
+    METHOD_WEBSOCKET,
+    RESULT_AUTH_MISSING,
+    RESULT_NOT_SUCCESSFUL,
+    RESULT_NOT_SUPPORTED,
+    RESULT_SUCCESS,
+    VALUE_CONF_ID,
+    VALUE_CONF_NAME,
+)
+
+
+class SamsungTVBridge(ABC):
+    """The Base Bridge abstract class."""
+
+    @staticmethod
+    def get_bridge(config):
+        """Get Bridge instance."""
+        if config:
+            if config[CONF_METHOD] == METHOD_LEGACY:
+                return SamsungTVLegacyBridge(config)
+            if config[CONF_METHOD] == METHOD_WEBSOCKET:
+                return SamsungTVWSBridge(config)
+        return None
+
+    def __init__(self, config):
```

Comment:
I'm thinking we should specify the parameters that are required and allowed here, by using positional and keyword arguments. Then create the config inside this bridge. Otherwise we may up in an inconsistent state where the user of the bridge passes in incomplete config.

Reply:
I do not understand, sorry.

Reply:
I suggest we internalize the config to this class since it seems part of the class responsibilities. It will be easier to guarantee that the correct config is always created.

We specify explicitly what parameters are required when creating the class instead of accepting an arbitrary dictionary.